### PR TITLE
[itests mqtt.homeassistant] Fixed itests for mqtt.homeassistant

### PR DESCRIPTION
**Pull request by mehoekstra**
While I have been making a few small bugfixes to binding.mqtt.homeassistant, it is unsatisfying not to see the green checkmark on the Travis integration tests :).  I have made the following updates to allow the integration tests to run.

Fixed the script buildci.sh.  The sed pattern for CHANGED_ITEST_DIR was not quite correct, if two directories were changed in the same test, then a build of all was forced.  With my change, with this current pull request everything will be built for the CI tests, but if someone approves this pull request, in future pull requests just for the tests, only the modified test will build and be tested.

Bigger overall change, changed the settings for bnd-testing-maven-plugin in itests/pom.xml to do OSGi resolution on the Travis machine before running the tests.  With this change, the -runbundles section in the itest.bndrun files are effectively ignored, and automatically regenerated before the test runs.  This change appears to take care of errors due to different environments between dev systems and the CI system.  Along with this change, the settings for bnd-export-maven-plugin in openhab-addons/pom.xml were changed as not to fail if the OSGi resolution on the CI machine computes different dependencies than are listed in the itest.bndrun files.

To make sure that all of the dependencies are available for the OSGi resolution, the buildci.sh script now also builds the  bom/*-index projects before building the tests.

Added missing dependencies for com.google.dagger to the three mqtt related itests.

Added biz.aQute.tester to itests.bndrun for mqtt.homeassistant.tests.  With this change, the integration tests can easily be run and debugged in Eclipse.

Changed the version of bndtools in openhab-addons/pom.xml to grab the latest 5.0.0 version.  This showed fewer errors in my testing.

With the changes in this pull request.  org.openhab.binding.mqtt.homeassistant.tests successfully builds and runs.  The other MQTT related tests successfully launch now, but the tests fail in execution, I did not debug those so I left them commented out in the openhab-addons/itests/pom.xml file.

### CODE REVIEW

**Comment by mehoekstra** [comment]
NEARLY achieved the elusive green check mark on the Travis tests, only the Hue emulation service test failed for JDK11.  Still, if someone approves this pull request one could work on the hue emulation test and only force a rebuild/test of that test.

**Comment by mehoekstra** [comment]
3rd time is a charm hopefully, got a timeout in the JDK8 test, tried increasing wait time.

**Comment by J-N-K** [comment]
Thanks. I‘ll have a deeper look later. Do you have an idea why the broker tests fail? I tried to figure that out for several hours and failed (probably looked in the wrong place).

**Comment by mehoekstra** [comment]
> Thanks. I‘ll have a deeper look later. Do you have an idea why the broker tests fail? I tried to figure that out for several hours and failed (probably looked in the wrong place).

@J-N-K, I haven't really looked at it yet, I was just happy that the integration test started running for me :).  I will try to take a look over the next couple of days.

**Comment by mehoekstra** [comment]
SUCCESS!  The Travis integration tests pass.

File: buildci.sh [hunk 1]
```diff
@@ -62,11 +62,45 @@ if [[ ! -z "$CHANGED_DIR" ]] && [[ -e "bundles/$CHANGED_DIR" ]]; then
...
+          stdbuf -o0 grep -v "target/code-analysis" | # filter out some debug code from reporting utility
+          tee -a ${CDIR}/.build.log
+        if [[ $? -ne 0 ]]; then
+            exit 1
+        fi
+         
+        # build the runtime index project so the OSGI index.xml file is generated in the target directory
+        # needed for resolving the OSGI bundle wiring later
+        cd "../runtime-index"
+        mvn clean install -B 2>&1 |
```
Comment (wborn) [inline buildci.sh]: Can you extracting the build commands to a function instead of duplicating them 4x?

Comment (wborn) [reply buildci.sh]: Or even better use a project list so they are all build in the same reactor using:

`-pl :org.openhab.addons.bom.openhab-core-index,:org.openhab.addons.bom.runtime-index,:org.openhab.addons.bom.test-index`

You can probably also add `:$CHANGED_DIR.tests:` to that list

Comment (mehoekstra) [reply buildci.sh]: Good point, the -pl flag is a good idea, will switch to that.

Comment (mehoekstra) [reply buildci.sh]: I applied the recommended changes, the script now uses the mvn -pl flag.

**Comment by mehoekstra** [comment]
@wborn and @J-N-K can I get an approval?  I'm not planning to add anything else to this pull request.  I'd like to get it in before I finalize by other mqtt.homeassistant enhancement #7080 so I can make sure that one doesn't break the integration test.  I'll also look more closely at the mqtt.homeassistant integration test and see if I can add a test for the change in #7080.  And I can take a look at why the other MQTT integration tests aren't passing, now that they run.

**Comment by kaikreuzer** [comment]
@mehoekstra This seems to have broken our build, see https://ci.openhab.org/view/Integration%20Builds%20(2.5.x)/job/openHAB2.5.x-Addons/

Could you have a look?

**Comment by kaikreuzer** [comment]
@mehoekstra This seems to have broken our build, see https://ci.openhab.org/view/Integration%20Builds%20(2.5.x)/job/openHAB2.5.x-Addons/

Could you have a look?

**Pull request merged**